core/domain/state_domain.py

```diff
@@ -822,13 +822,23 @@ def from_dict(cls, content_translations_dict):
 
         return cls(content_translations)
 
-    def validate(self):
+    def validate(self, expected_content_id_list=None):
```

Comment:
Why is this an optional arg? Can it be mandatory?

If it is because of skew between questions and explorations, then the check which needs this should be done at the Exploration level and not the State level.